### DISCUSSION TITLE
Add literal from content of a file with newline

### DISCUSSION THREAD

**Discussion by saadlu** [mikefarah/yq#713] [Q&A] [answered]
Hi helpful ppl,

Say I want create the following yaml

```
a: |
  this is line 1
  this is line 2
```

where them lines are coming from a file, `file.txt`

```txt
  this is line 1
  this is line 2
```

is this possible?

**Comment by saadlu** [comment]
This is what I can come up with for now. Using `jq` in the process

```sh
echo '{"a":null}' | jq --arg e "$(cat file.txt)" '.a = $e' |  yq e -P -
```

I am sure this can be improved upon :)

**Comment by mikefarah** [comment, marked as answer]
```
content=$(cat file.txt) yq e -n '.a = strenv(content)'
```

**Reply by saadlu** [reply]
nice one. I tried `env` operator, but did not read about `strenv`. 

(now I read the manual ;P)

**Comment by mikefarah** [comment]
And if you want to keep the trailing newlines then:

```
content="$(cat file.txt)
" yq e -n '.a = strenv(content)'
```

(Note that this is a two line command)

**Comment by BenCornwell1** [comment]
I was having trouble with this issue, merging into an existing yaml.  I had to paste a PEM into a Kubernetes CR.  It worked fine in yq3.

I originally had this:

`export message=$(cat multiline.txt)
yq eval -i '.test.message = env(message)' test.yaml`

Using the export command.  It worked, but it put my multi-line content all on one line, rather than a yaml multi-line block.  As above - what worked was this:

`message=$(cat multiline.txt) yq eval -i '.test.message = strenv(message)' test.yaml`

Can I suggest this goes in the documentation?
